pkg/k8s/object/builders/agent/secrets/config.go

```diff
@@ -122,17 +122,24 @@ func (c *configBuilder) backendConfig() (map[string][]byte, error) {
 
 	// render additional backends configuration
 	for i, backend := range c.Spec.Agent.AdditionalBackends {
-		if backend.Key == "" || backend.EndpointHost == "" {
+		if (c.Spec.Agent.KeysSecret == "" && backend.Key == "") || backend.EndpointHost == "" {
 			// skip backend as it would be broken anyways, should be caught by the schema validator anyways, but better be safe than sorry
-			c.logger.Error(fmt.Errorf("key or endpointHost undefined"), "skipping additional backend due to missing values")
+			c.logger.Error(fmt.Errorf("backend not defined: backend key (plaintext or through secret) or endpoint missing"), "skipping additional backend due to missing values")
 			continue
 		}
 
+		var backendKey string
+		if keyValueFromSecret, ok := c.keysSecret.Data["key"]; ok {
```

Comment:
Close, but you can't always use the first key, you need to consider the position of the backend I think.
As is, it will use the first key for all backends:

```
[root@worker2 agent]# cat ./etc/instana/com.instana.agent.main.sender.Backend-2.cfg
host=second-backend.instana.io
port=443
protocol=HTTP/2
key=xxx
[root@worker2 agent]# cat ./etc/instana/com.instana.agent.main.sender.Backend-1.cfg
host=first-backend.instana.io
port=443
protocol=HTTP/2
key=xxx

```